jbmc/src/java_bytecode/convert_java_nondet.cpp

```diff
@@ -114,25 +147,36 @@ static goto_programt::targett insert_nondet_init_code(
 /// \param goto_program: The goto program to modify.
 /// \param symbol_table: The global symbol table.
 /// \param message_handler: Handles logging.
-/// \param max_nondet_array_length: Maximum size of new nondet arrays.
+/// \param object_factory_parameters: Parameters for the generation of nondet
+///   objects.
+/// \param mode: Language mode
 void convert_nondet(
   goto_programt &goto_program,
   symbol_table_baset &symbol_table,
   message_handlert &message_handler,
   const object_factory_parameterst &object_factory_parameters,
   const irep_idt &mode)
 {
-  for(auto instruction_iterator=goto_program.instructions.begin(),
-        end=goto_program.instructions.end();
-      instruction_iterator!=end;)
+  bool changed = false;
+  auto instruction_iterator = goto_program.instructions.begin();
+  auto end = goto_program.instructions.end();
```

Comment:
These hand-optimisations can actually make things worse. Leave end in the comparison below. There are Stack Overflow articles about this.